tests/test_views.py

```diff
@@ -86,3 +96,103 @@ def test_post_detail_failed(self):
             reverse('post_detail', kwargs={'pk': 31}),
         )
         self.assertEqual(404, http_response.status_code)
+
+    def test_post_new_get_authorized(self):
+        """Ensure authorized users can access the new post page."""
+        self.client.force_login(self.user)
+        url = reverse('post_new')
+        http_response = self.client.get(url)
+        self.assertEqual(http_response.status_code, 200)
+        self.assertTemplateUsed(http_response, 'blog/post_edit.html')
+
+    def test_post_new_get_unauthorized(self):
+        """Ensure unauthorized users are redirected when add the new post."""
+        self.client.logout()
+        url = reverse('post_new')
+        http_response = self.client.get(url)
+        self.assertRedirects(http_response, f'/accounts/login/?next={url}')
+
+    def test_post_new_valid_form_authorized(self):
+        """Ensure authorized users can correctly fill out the form for post."""
+        self.client.force_login(self.user)
+        url = reverse('post_new')
+        initial_post_count = Post.objects.count()
+        http_response = self.client.post(
+            url, self.valid_form_post, follow=True,
+        )
+        self.assertEqual(http_response.status_code, 200)
+        self.assertTemplateUsed(http_response, 'blog/post_detail.html')
+        new_post = Post.objects.last()
+        self.assertRedirects(
+            http_response,
+            reverse(
+                'post_detail',
+                kwargs={'pk': new_post.pk},
+            ),
+        )
+        self.assertEqual(new_post.title, 'Test Title')
+        self.assertEqual(new_post.text, 'Test Text')
+        self.assertEqual(new_post.author, self.user)
+        self.assertIsNotNone(new_post.published_date)
+        self.assertEqual(Post.objects.count(), initial_post_count + 1)
+
+    def test_post_new_invalid_form_authorized(self):
+        """Test handling of invalid form submission for new post."""
+        self.client.force_login(self.user)
+        url = reverse('post_new')
+        http_response = self.client.post(url, self.invalid_form_post)
+        self.assertEqual(http_response.status_code, 200)
+        self.assertTemplateUsed(http_response, 'blog/post_edit.html')
+        self.assertContains(
+            http_response, 'This field is required.', html=True,
+        )
```

Comment:
Try using the `assertFormError()` helper:
- https://docs.djangoproject.com/en/5.1/topics/testing/tools/#django.test.SimpleTestCase.assertFormError
- https://developer.mozilla.org/en-US/docs/Learn/Server-side/Django/Testing#testing_views_with_forms
- https://gist.github.com/cwurld/f132f673446158808952#file-django_testcase_cheatsheet-py-L80